chaoskube/chaoskube.go

```diff
@@ -62,11 +67,32 @@ func New(client kubernetes.Interface, labels, annotations, namespaces labels.Sel
 func (c *Chaoskube) Candidates() ([]v1.Pod, error) {
 	listOptions := metav1.ListOptions{LabelSelector: c.Labels.String()}
 
-	podList, err := c.Client.Core().Pods(v1.NamespaceAll).List(listOptions)
+	podList, err := c.Client.CoreV1().Pods(v1.NamespaceAll).List(listOptions)
 	if err != nil {
 		return nil, err
 	}
 
+	if !c.NamespaceLabels.Empty() {
+		nsListOptions := metav1.ListOptions{LabelSelector: c.NamespaceLabels.String()}
+		namespacesItems, err := c.Client.CoreV1().Namespaces().List(nsListOptions)
+		if err != nil {
+			return nil, err
+		}
+
+		// Extract matching namespaces
+		namespaceList := []string{}
+		for _, ns := range namespacesItems.Items {
+			namespaceList = append(namespaceList, ns.Name)
+		}
+		namespaceLabel, _ := labels.Parse(strings.Join(namespaceList, ","))
+
+		// Append additional namespaces filters
+		reqs, _ := namespaceLabel.Requirements()
+		for _, req := range reqs {
+			c.Namespaces = c.Namespaces.Add(req)
```

Comment:
I have to think about this. I was thinking we could do this in the initializer (resolving the namespace label selector and then adding the matched namespaces to the name-based namespace selector).

But then, having it here makes much more sense as the label selector is re-evaluated at every loop (namespaces could have been created, labels on namespaces can chenge to opt-in them etc.). However, modifying the chaoskube-instance global variable here might not have the desired effect in this case.